src/pages/resultsView/ResultsViewPageStore.ts

```diff
@@ -1630,6 +1633,112 @@ export class ResultsViewPageStore {
+                samples = this.samples.result.map(sample => {
+                    return {
+                        molecularProfileId: sample.studyId + '_structural_variants',
+                        sampleId: sample.sampleId
+                    };
+                });
+                params.sampleMolecularIdentifiers = samples;
+            } else {
+                // get molecular profile ids
+                if (this.selectedMolecularProfiles.result) {
+                    molecularProfiles = this.selectedMolecularProfiles.result.filter(profile => {
+                        return profile.molecularAlterationType === AlterationTypeConstants.STRUCTURAL_VARIANT;
+                    }).map(profile => {
+                        return profile.molecularProfileId;
+                    });
+                    params.molecularProfileIds = molecularProfiles;
+                }
+            }
+            return await client.fetchStructuralVariantsUsingPOST({'structuralVariantFilter':<StructuralVariantFilter> params});
+        }
+    });
+
+    /**
+     * Group fusion data by genes.
+     * @returns {{[p: string]: StructuralVariant[]}}
+     */
+    @computed
+    get fusionsByGene(): { [hugeGeneSymbol: string]: StructuralVariant[] } {
+        let genes:Array<string> = [];
+        let fusionsByGene:{ [hugeGeneSymbol: string]: StructuralVariant[] } = {};
+        // get string values of genes from input query
+        if (this.genes.result) {
+            // get the genes
+            genes = this.genes.result.map(gene => gene.hugoGeneSymbol);
+            // group fusion data by the genes from input query
+            genes.forEach(gene => {
+                let tmpFusions:Array<any> = [];
+                if (this.fusions.result) {
+                    tmpFusions = tmpFusions.concat(this.fusions.result.filter((fusion:StructuralVariant) => {
+                        return fusion.site1HugoSymbol === gene || fusion.site2HugoSymbol === gene;
+                    }));
+                    fusionsByGene = Object.assign(fusionsByGene, { [gene]: tmpFusions });
+                }
+            });
+        }
+        return fusionsByGene;
+    }
+
+    /**
+     * Assign a FusionMapperStore per genes
+     * @type {MobxPromiseUnionTypeWithDefault<{[p: string]: FusionMapperStore}>}
+     */
+    readonly fusionMapperStores = remoteData<{[hugoGeneSymbol: string]: ResultViewFusionMapperStore}>({
+        invoke: () => {
+            if (this.genes.result && this.fusionsByGene) {
+                return Promise.resolve(
+                    this.genes.result.reduce((map: { [hugoGeneSymbol: string]: ResultViewFusionMapperStore }, gene: Gene) => {
+                        map[gene.hugoGeneSymbol] = new ResultViewFusionMapperStore(
+                            gene,
+                            this.studyIdToStudy,
+                            this.molecularProfileIdToMolecularProfile,
+                            this.samples,
```

Comment:
all of these need to be awaited - samples, molecularProfileIdToMolecularProfile, studyIdToStudy.